src/game.cpp

```diff
@@ -9144,8 +9144,20 @@ bool game::walk_move( const tripoint &dest_loc )
     }
     u.set_underwater( false );
 
-    if( !shifting_furniture && !pushing && !prompt_dangerous_tile( dest_loc ) ) {
-        return true;
+
+    if( !shifting_furniture && !pushing && is_dangerous_tile( dest_loc ) ) {
+        if( !u.movement_mode_is( CMM_RUN ) ) {
+            std::vector<std::string> harmful_stuff = get_dangerous_tile( dest_loc );
+            std::string key_message = press_x( ACTION_TOGGLE_RUN );
+            add_msg( m_warning,
+                     _( "Stepping into that %1$s looks risky. %2$s to run into it to force movement." ),
```

Comment:
The verbiage in this warning is only correct if `DANGEROUS_RUNNING` is false, but you don't check for it here.

Further, `ACTION_TOGGLE_RUN` might not be defined for a player.

Reply:
Given the number of ways someone might have their bindings set up for running (there's three completely different ways to set movement mode, and only cycling is bound by default), displaying a key here is probably not the way to go.